packages/ui/src/components/button/button.vue

```diff
@@ -127,245 +124,371 @@
+        var(--acv-color-button-disabled)
       );
-      --acv-button-text-color: var(--acv-color-text-inversed-primary);
-    }
-
-    &.outline {
-      --acv-button-background-color-active: hsl(from var(--acv-button-color) h s calc(l + 30));
-      --acv-button-background-color-disabled: var(--acv-button-background-color);
-      --acv-button-background-color-hover: hsl(from var(--acv-button-color) h s calc(l + 30));
-      --acv-button-background-color: var(--acv-color-transparent);
-      --acv-button-text-color: var(--acv-button-color);
-      --acv-button-text-color-disabled: hsl(from var(--acv-button-text-color) h s calc(l + 10));
-      --acv-button-text-color-hover: hsl(from var(--acv-button-text-color) h s calc(l - 10));
-      --acv-button-text-color-active: hsl(from var(--acv-button-text-color) h s calc(l - 20));
-      --acv-button-border-color-disabled: hsl(from var(--acv-button-color) h s calc(l + 10));
-      --acv-button-border-color-hover: hsl(from var(--acv-button-color) h s calc(l - 10));
-      --acv-button-border-color-active: hsl(from var(--acv-button-color) h s calc(l - 30));
-      --acv-button-border-color: var(--acv-button-color);
-      --acv-button-border-width: var(--acv-border-regular);
-    }
-
-    &.ghost {
-      --acv-button-background-color-active: hsl(from var(--acv-button-color) h calc(s - 12) calc(l + 23));
-      --acv-button-background-color-disabled: hsl(from var(--acv-button-background-color) h s calc(l + 10));
-      --acv-button-background-color-hover: hsl(from var(--acv-button-color) calc(h - 1) calc(s - 11) calc(l + 33));
-      --acv-button-background-color: var(--acv-color-transparent);
-      --acv-button-border-width: var(--acv-base-border-width-00);
-      --acv-button-text-color: var(--acv-button-color);
-      --acv-button-text-color-active: hsl(from var(--acv-button-color) h s calc(l - 20));
-      --acv-button-text-color-disabled: hsl(from var(--acv-button-color) h s calc(l - 10));
-      --acv-button-text-color-hover: hsl(from var(--acv-button-color) h s calc(l - 10));
-    }
-
-    &.light {
-      --acv-button-background-color-active: hsl(
-          from var(--acv-button-background-color) calc(h - 5) calc(s + 5)
-          calc(l - 20)
+      --_acv-button-text-color-hover: var(
+        --acv-button-primary-text-color-hover,
+        var(--acv-color-button-label-inversed-primary)
       );
-      --acv-button-background-color-disabled: hsl(
-          from var(--acv-button-background-color) h s calc(l + 10)
+      --_acv-button-text-color-active: var(
+        --acv-button-primary-text-color-active,
+        var(--acv-color-button-label-inversed-primary)
       );
-      --acv-button-background-color-hover: hsl(
-          from var(--acv-button-background-color) h s calc(l - 10)
+      --_acv-button-text-color-disabled: var(
+        --acv-button-primary-text-color-disabled,
+        var(--acv-color-button-label-disabled-inversed)
```

Comment:
the more I think about this approach the more I find downsides. Probably it's better to leave just one css variable
```suggestion
      --_acv-button-text-color-disabled: var(--acv-button-primary-text-color-disabled)
```

I mean rename `acv-color-button-label-disabled-inversed` to `acv-button-primary-text-color-disabled`. And make the same with other vars